x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/CompositeBucketsChangeCollector.java

```diff
@@ -175,7 +227,10 @@ public boolean collectChanges(Collection<? extends Bucket> buckets) {
 
         @Override
         public QueryBuilder filterByChanges(long lastCheckpointTimestamp, long nextcheckpointTimestamp) {
+
             if (isSynchronizationField && lastCheckpointTimestamp > 0) {
+                // missing bucket can't be enabled as it would be an illegal combination
```

Comment:
Is this enforced anywhere in the config parsing?

And do the `randomTransformConfig` methods in the tests enforce this?  I guess if they choose completely random field names for each section of the config then it's highly unlikely the time field in the sync config would randomly be the same as the time field for the date histogram.  So maybe this problem can be ignored.

Reply:
good spot!

I should instead disable the `isSynchronizationField` optimization on line 214 in case `missingBucket == true`:

```
this.isSynchronizationField = isSynchronizationField && (missingBucket == false);
```